### PR TITLE
Keybindings: support time range copy/paste 

### PR DESCRIPTION
**Pull request by gtk-grafana**
Add support for `t c` and `t v` in Explore Logs.

Related: https://github.com/grafana/scenes/pull/999
Also ran into this bug while testing: https://github.com/grafana/grafana/issues/97892

### CODE REVIEW

File: src/Components/IndexScene/IndexScene.tsx [hunk 1]
```diff
@@ -221,6 +222,33 @@ export class IndexScene extends SceneObjectBase<IndexSceneState> {
...
+      } else {
+        logger.error(new Error('Invalid time range from clipboard'), {
+          msg: 'Invalid time range from clipboard',
+          sceneTimeRange: typeof timeRange,
+          to: to ?? '',
+          from: from ?? '',
+        });
+      }
+    }
+  };
```
Comment (matyax) [inline src/Components/IndexScene/IndexScene.tsx:250]: As usual, one less level of indentation.

```suggestion
  private subscribeToPasteTimeEvent = async () => {
    const copiedRange = await getCopiedTimeRange();

    if (!copiedRange.isError) {
      return;
    }
    const timeRange = sceneGraph.getTimeRange(this);
    const to = typeof copiedRange.range.to === 'string' ? copiedRange.range.to : undefined;
    const from = typeof copiedRange.range.from === 'string' ? copiedRange.range.from : undefined;

    const newRange = rangeUtil.convertRawToRange(copiedRange.range);

    if (timeRange && newRange) {
      timeRange.setState({
        value: newRange,
        to,
        from,
      });
    } else {
      logger.error(new Error('Invalid time range from clipboard'), {
        msg: 'Invalid time range from clipboard',
        sceneTimeRange: typeof timeRange,
        to: to ?? '',
        from: from ?? '',
      });
    }
  };
```

File: src/services/keyboardShortcuts.ts [hunk 2]
```diff
@@ -63,6 +64,26 @@ export function setupKeyboardShortcuts(scene: IndexScene) {
     }),
   });
 
+  // Copy time range
+  keybindings.addBinding({
+    key: 't c',
+    onTrigger: () => {
+      const timeRange = sceneGraph.getTimeRange(scene);
+      setWindowGrafanaSceneContext(timeRange);
```
Comment (matyax) [inline src/services/keyboardShortcuts.ts:72]: Not entirely sure what this is doing.

File: src/services/keyboardShortcuts.ts [hunk 3]
```diff
@@ -146,3 +167,64 @@ export function toggleVizPanelLegend(vizPanel: VizPanel): void {
...
+      (window as any).__grafanaSceneContext = prevScene;
+    }
+  };
+}
+
+// taken from /Users/galen/projects/grafana/grafana/public/app/core/utils/timePicker.ts
+type CopiedTimeRangeResult = { range: RawTimeRange; isError: false } | { range: string; isError: true };
+
+// modified to narrow types from clipboard
+export async function getCopiedTimeRange(): Promise<CopiedTimeRangeResult> {
```
Comment (matyax) [inline src/services/keyboardShortcuts.ts:210]: A couple of unit tests wouldn't hurt.

File: src/services/keyboardShortcuts.ts (hunk 2 above)
Comment (gtk-grafana) [reply src/services/keyboardShortcuts.ts:72]: you are not alone :P 
I added the docblock from the definition in scenes, the important bit `This is also used from TimeSrv to access scene time range.`

**Review by matyax — APPROVED** [review]
💯

**Pull request merged**